apps/hash-graph/lib/graph/src/api/rest/entity_type.rs

```diff
@@ -93,29 +102,75 @@ async fn create_entity_type<P: StorePool + Send>(
     pool: Extension<Arc<P>>,
     domain_validator: Extension<DomainValidator>,
     body: Json<CreateEntityTypeRequest>,
-) -> Result<Json<OntologyElementMetadata>, StatusCode> {
+    // TODO: We want to be able to return `Status` here we should try and create a general way to
+    //  call `status_to_response` for our routes that return Status
+) -> Result<Json<OntologyElementMetadata>, Response> {
     let Json(CreateEntityTypeRequest {
         schema,
         owned_by_id,
         actor_id,
     }) = body;
 
-    let entity_type: EntityType = schema.try_into().into_report().map_err(|report| {
-        tracing::error!(error=?report, "Couldn't convert schema to Entity Type");
-        // Shame there isn't an UNPROCESSABLE_ENTITY_TYPE code :D
-        StatusCode::UNPROCESSABLE_ENTITY
-        // TODO - We should probably return more information to the client
-        //  https://app.asana.com/0/1201095311341924/1202574350052904/f
+    let entity_type: EntityType = schema.try_into().map_err(|err: ParseEntityTypeError| {
+        tracing::error!(error=?err, "Provided schema wasn't a valid entity type");
+        status_to_response(Status::new(
+            hash_status::StatusCode::InvalidArgument,
+            Some("Provided schema wasn't a valid entity type.".to_owned()),
+            vec![StatusPayloads::ErrorInfo(ErrorInfo::new(
+                HashMap::from([(
+                    "validationError".to_owned(),
+                    serde_json::to_value(err)
+                        .expect("Could not serialize entity type validation error"),
+                )]),
+                // TODO: We should encapsulate these Reasons within the type system, perhaps
+                //  requiring top level contexts to implement a trait `ErrorReason::to_reason`
+                //  or perhaps as a big enum, or as an attachment
+                "INVALID_SCHEMA".to_owned(),
+            ))],
```

Comment:
I guess these should be provided by `error-stack` at some point?

Reply:
I think that'd be good, yes